sticky-custom-post-types/sticky-custom-post-types.php

```diff
@@ -25,84 +36,263 @@
-}
+			return $classes;
+		}
 
-function super_sticky_admin_init() {
-	register_setting('reading', 'sticky_custom_post_types');
-	register_setting('reading', 'sticky_custom_post_types_filters');
-	add_settings_section('super_sticky_options', 'Sticky Custom Post Types', 'super_sticky_description', 'reading');
-	add_settings_field('sticky_custom_post_types', 'Show "Stick this..." checkbox on', 'super_sticky_set_post_types', 'reading', 'super_sticky_options');
-	add_settings_field('sticky_custom_post_types_filters', 'Display selected post type(s) on', 'super_sticky_set_filters', 'reading', 'super_sticky_options');
-}
+		/**
+		 * Add sticky settings.
+		 *
+		 * @return void
+		 */
+		public function add_sticky_settings() {
+			register_setting( 'reading', 'sticky_custom_post_types' );
+			register_setting( 'reading', 'sticky_custom_post_types_filters' );
 
-add_action('admin_init', 'super_sticky_admin_init', 20);
+			add_settings_section(
+				'super_sticky_options',
+				__( 'Sticky Custom Post Types', 'sticky-custom-post-types' ),
+				[ $this, 'sticky_description' ],
+				'reading'
+			);
 
-function super_sticky_post_types() {
-	return (array) get_option( 'sticky_custom_post_types', array() );
-}
+			add_settings_field(
+				'sticky_custom_post_types',
+				__( 'Show "Stick this..." checkbox on', 'sticky-custom-post-types' ),
+				[ $this, 'set_sticky_post_types' ],
+				'reading',
+				'super_sticky_options'
+			);
 
-function super_sticky_meta() { global $post; ?>
-	<input id="super-sticky" name="sticky" type="checkbox" value="sticky" <?php checked(is_sticky($post->ID)); ?> /> <label for="super-sticky" class="selectit"><?php _e('Stick this to the front page') ?></label><?php
-}
+			add_settings_field(
+				'sticky_custom_post_types_filters',
+				__( 'Display selected post type(s) on', 'sticky-custom-post-types' ),
+				[ $this, 'set_sticky_filters' ],
+				'reading',
+				'super_sticky_options'
+			);
+		}
 
-function super_sticky_add_meta_box() {
-	foreach(super_sticky_post_types() as $post_type)
-		if(current_user_can('edit_others_posts'))
-			add_meta_box('super_sticky_meta', 'Sticky', 'super_sticky_meta', $post_type, 'side', 'high');
-}
+		/**
+		 * Sticky Description.
+		 *
+		 * @return void
+		 */
+		public function sticky_description() {
+			echo '<p>' . esc_html__( 'Enable support for sticky custom post types.', 'sticky-custom-post-types' ) . '</p>'; // WPCS: XSS ok.
+		}
+
+		/**
+		 * Set sticky options.
+		 *
+		 * @return void
+		 */
+		public function set_sticky_post_types() {
+			$post_types = get_post_types( [
+				'_builtin' => false,
+				'public'   => true,
+			], 'names' );
 
-add_action('admin_init', 'super_sticky_add_meta_box');
+			if ( empty( $post_types ) ) {
+				echo '<p>' . esc_html__( 'No public custom post types found.', 'sticky-custom-post-types' ) . '</p>'; // WPCS: XSS ok.
+			}
+
+			// Filter the post types supported.
+			$post_types = apply_filters( 'sticky_supported_custom_post_types', $post_types );
```

Comment:
This adds a way for developers to remove certain **valid** custom post types programatically.